### PR TITLE
Feature/pz 23 implementation of community grid

### PR DESCRIPTION
**Pull request by Prashishgt**
# Merge Request

## Summary of change

image
image
image
image

## Mention

Review: @dipeshkumarsah98 @sarojghalan @subhashishjungshah 

Fixes: PZ-23- Implementation of Community Grid

## Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that would cause existing functionality to not work as expected)
- [ ] Front End (UI) changes
- [ ] Changes in `documentation`

## How did you test

- [X] Rendered in community page. /community

## Checklist

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [X] My code follows the style guidelines of this project
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added required tests that prove my fix is effective or that my feature works
- [ ] Both New and Existing unit tests pass locally with my changes

### CODE REVIEW

File: components/PZCard/PZModalProductCard/index.tsx [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+  StyledPZCommunityModalCard,
+  StyledPZCommunityModalCardInformation,
+} from "./style";
+import { PZCommunityModalProductCardPropsI } from "./types";
+
+const PZCommunityModalProductCard = ({
+  cardImg,
+}: PZCommunityModalProductCardPropsI) => (
+  <StyledPZCommunityModalCard>
+    <Image
```
Comment (sarojghalan) [inline components/PZCard/PZModalProductCard/index.tsx:15]: as other MUI component, wrap this Image of next in styled component.

File: components/PZCommunity/PZCommunityModal/index.tsx [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+  onClose,
+  imageSrc,
+  onNext,
+  onPrev,
+  ...props
+}: PZCommunityModalPropsI) => (
+  <PZModal onClose={onClose} {...props}>
+    <StyledPZCommunityModalWrapper>
+      <StyledPZCommunityModalImage>
+        <Image src={imageSrc} alt="default" objectFit="cover" />
```
Comment (sarojghalan) [inline components/PZCommunity/PZCommunityModal/index.tsx:82]: Wrap this Image in styled component and override from there.

File: components/PZCommunity/PZCommunityGrid/index.tsx [hunk 3]
```diff
@@ -0,0 +1,71 @@
+"use client";
+import Image from "next/image";
+import { useState } from "react";
+import PZButton from "@/components/PZButton";
+import PZCommunityModal from "@/components/PZCommunity/PZCommunityModal";
+import { dummyImg } from "@/data/dummyImageforCommunityMoal";
+import { StyledPZCommunityGrid, StyledPZCommunityGridItem } from "./style";
+
+const PZCommunityGridItem = ({ hasSrc, handleClick }: any) => (
```
Comment (sarojghalan) [inline components/PZCommunity/PZCommunityGrid/index.tsx:9]: why there is any data types? writing any is not better approach.

File: components/PZCommunity/PZCommunityModal/types.ts [hunk 4]
```diff
@@ -0,0 +1,14 @@
+export type PZCommunityModalPropsI = {
+  open: boolean;
+  onClose: () => void;
+  onNext: () => void;
+  onPrev: () => void;
+  imageSrc: any;
```
Comment (sarojghalan) [inline components/PZCommunity/PZCommunityModal/types.ts:6]: there is data type of next for images, replace any with that

File: components/PZCommunity/PZCommunityModal/types.ts [hunk 5]
```diff
@@ -0,0 +1,14 @@
...
+};
+
+export interface PZCommunityModalRightItemsPropsI {
+  onClose: () => void;
+  onNext: () => void;
+  onPrev: () => void;
+  src: any;
```
Comment (sarojghalan) [inline components/PZCommunity/PZCommunityModal/types.ts:13]: same replace it with `next` image data type.

**Review by sarojghalan — CHANGES_REQUESTED** [review]
Nice , Keep up making progress.

File: components/PZCommunity/PZCommunityGrid/style.ts [hunk 6]
```diff
@@ -0,0 +1,87 @@
+"use client";
+import { Grid, Box } from "@mui/material";
+import styled from "styled-components";
+
+export const StyledPZCommunityGrid = styled(Grid)`
```
Comment (subhashishjungshah) [inline components/PZCommunity/PZCommunityGrid/style.ts:5]: You can also use material UI's Image list in future. It is easy than this

**Review by subhashishjungshah — CHANGES_REQUESTED** [review]
Good work!  Make some changes.

**Pull request closed without merging**